src/Raven.Server/Rachis/ClusterMaintenanceSlave.cs

```diff
@@ -0,0 +1,135 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Raven.Client.Documents.Replication.Messages;
+using Raven.Server.Documents.TcpHandlers;
+using Raven.Server.ServerWide;
+using Raven.Server.ServerWide.Context;
+using Sparrow;
+using Sparrow.Json;
+using Sparrow.Json.Parsing;
+
+namespace Raven.Server.Rachis
+{
+    public class ClusterMaintenanceSlave : IDisposable
+    {
+        private readonly TcpConnectionOptions _tcp;
+        private readonly ServerStore _server;
+        private CancellationToken _token;
+        private readonly CancellationTokenSource _cts;
+        private Task _collectingTask;
+        private Dictionary<string,DatabaseStatusReport> _cachedReport = new Dictionary<string, DatabaseStatusReport>();
+
+        public ClusterMaintenanceSlave(TcpConnectionOptions tcp, CancellationToken externalToken, ServerStore serverStore)
+        {
+            _tcp = tcp;
+            _cts = CancellationTokenSource.CreateLinkedTokenSource(externalToken);
+            _token = _cts.Token;
+            _server = serverStore;
+        }
+
+        public void Start()
+        {
+            _collectingTask = CollectReport();
+        }
+
+        public async Task CollectReport()
+        {
+            while (_token.IsCancellationRequested == false)
+            {
+                try
+                {
+                    using (_server.ContextPool.AllocateOperationContext(out TransactionOperationContext ctx))
+                    using (ctx.OpenReadTransaction())
+                    {
+                        var dbs = CollectDatabaseInformation(ctx);
+                        using (var writer = new BlittableJsonTextWriter(ctx, _tcp.Stream))
+                        {
+                            ctx.Write(writer, DynamicJsonValue.Convert(dbs.ToDictionary(k => k.Item1,v=>v.Item2)));
+                        }
+                    }
+                    
+                    await Task.Delay(TimeSpan.FromMilliseconds(500), _token);
+                }
+                catch (Exception)
+                {
+                    // TODO: log and abort
+                }
+            }
+        }
+
+        private IEnumerable<(string,DatabaseStatusReport)> CollectDatabaseInformation(TransactionOperationContext ctx)
```

Comment:
Why do we need a tuple? We already have it in the `Name` property of the `DatabaseStatusReport`.